src/store/reducers/common.js

```diff
@@ -238,41 +239,72 @@ export default (state = initialState, action) => {
         logInStatus: action.payload,
       };
     case 'SHOPLIST_SORT_PRODUCT': {
-      const { id } = action;
+      const { id, products: productData } = action;
       const { filterPrice } = state;
       const productState = state?.popularProducts;
-      const productSort = productState?.filter(
-        (prod) =>
-          filterPrice[0] <= prod.price &&
-          prod.price <= filterPrice[1] &&
-          (prod.category === id || prod.color === id)
-      );
+      let productSort;
+      if (productData) {
+        productSort = productData?.filter(
+          (prod) => filterPrice[0] <= prod.price && prod.price <= filterPrice[1]
+        );
+      } else {
+        productSort = productState?.filter(
+          (prod) =>
+            filterPrice[0] <= prod.price &&
+            prod.price <= filterPrice[1] &&
+            (prod.category === id || prod.color === id)
+        );
+      }
       return {
         ...state,
         shoplistSortProduct: productSort,
       };
     }
     case 'SHOPLIST_PRICE_FILTER': {
-      const { price, id } = action.value;
+      const { id, price, products: productData } = action;
       const productState = state?.popularProducts;
-      const productSort =
-        id === 'all'
-          ? productState?.filter(
-              (prod) => price[0] <= prod.price && prod.price <= price[1]
-            )
-          : productState?.filter(
-              (prod) =>
-                price[0] <= prod.price &&
-                prod.price <= price[1] &&
-                (prod.category.toString() === id ||
-                  prod.color.toString() === id)
-            );
+      let productSort;
+      if (!id) {
+        productSort = productData?.filter(
+          (prod) => price[0] <= prod.price && prod.price <= price[1]
```

Comment:
Just like above, consider splitting this into a function since we have too many places using the same code.